scripts/tools/TaskQueueMaintenance.php

```diff
@@ -0,0 +1,248 @@
+{
+    use ServiceLocatorAwareTrait;
+
+    private int $completedRetentionDays = 30;   // Completed/Failed -> Archived
+    private int $archivedRetentionDays  = 180;  // Archived -> Deleted
+    private int $stuckRetentionDays     = 14;   // In Progress / Enqueued -> “stuck”
+
+    /**
+     * Flags set from CLI arguments.
+     */
+    private bool $doArchive = false;
+    private bool $doUnblock = false;
+    private bool $doVacuum  = false;
+    private bool $showHelp  = false;
+
+    /**
+     * Entry point when called via:
+     *   sudo -u www-data php index.php 'oat\taoTaskQueue\scripts\tools\TaskQueueMaintenance'
+     *
+     * @param array $params CLI parameters passed by TAO
+     */
+    public function __invoke($params): Report
+    {
+        try {
+            $this->parseParams((array) $params);
+
+            $messages = [];
+
+            if ($this->doArchive) {
+                $count = $this->runArchive();
+                $messages[] = sprintf(
+                    '[TaskQueueMaintenance] Archive flow finished. Affected tasks: %d',
+                    $count
+                );
+            }
+
+            if ($this->doUnblock) {
+                // TODO: implement unblock flow
+                $messages[] = '[TaskQueueMaintenance] Unblock flow not implemented yet.';
+            }
+
+
+            if ($this->doVacuum) {
+                $this->runVacuum();
+                $messages[] = '[TaskQueueMaintenance] Vacuum flow finished (VACUUM FULL tq_task_log).';
+            }
+
+            // If no specific action requested, or --help given
+            if ($this->showHelp || (!$this->doArchive && !$this->doUnblock && !$this->doVacuum)) {
+                $messages[] = $this->commandOutput();
+            }
+
+            return Report::createSuccess(implode(PHP_EOL . PHP_EOL, $messages));
+        } catch (\Throwable $e) {
+            return Report::createError('[TaskQueueMaintenance] ' . $e->getMessage());
+        }
+    }
+
+    /**
+     * Argument parsing:
+     *  --archive
+     *  --unblock
+     *  --vacuum
+     *  --help
+     */
+    private function parseParams(array $params): void
+    {
+        foreach ($params as $param) {
+            switch ($param) {
+                case '--archive':
+                    $this->doArchive = true;
+                    break;
+
+                case '--unblock':
+                    $this->doUnblock = true;
+                    break;
+
+                case '--vacuum':
+                    $this->doVacuum = true;
+                    break;
+
+                case '--help':
+                    $this->showHelp = true;
+                    break;
+            }
+        }
+    }
+
+    /**
+     * Main entry point for the --archive option.
+     * Returns number of affected tasks.
+     */
+    private function runArchive(): int
+    {
+        $archivedCount = $this->archiveCompletedAndFailed();
```

Comment:
I expect Archive method only do Archive, and not Delete old archive. wdyt about Single Responsability pattern?